science/hdf5/Portfile

```diff
@@ -58,14 +58,19 @@ compiler.blacklist-append \
 compiler.c_standard 1999
 
 # Use lib/hdf5 rather than hdf5/lib plugin directory
-configure.args      --with-zlib=yes \
-                    --with-szlib=${prefix}/lib/libaec \
+configure.args \
+                    --disable-cxx \
+                    --disable-fortran \
+                    --disable-hl \
+                    --disable-parallel \
                     --disable-silent-rules \
-                    --enable-build-mode=production --disable-fortran \
-                    --disable-cxx --disable-hl --enable-shared --enable-static \
-                    --disable-parallel --disable-threadsafe \
+                    --disable-threadsafe \
+                    --enable-build-mode=production \
+                    --enable-shared \
+                    --enable-static \
                     --with-default-plugindir=${prefix}/lib/hdf5 \
-                    --with-default-api-version=v110
+                    --with-szlib=${prefix}/lib/libaec \
```

Comment:
For `--with-szlib`, was the intent to specify an invalid path, to sidestep opportunistic use of port `szip`?

Contents of the latter are:

```
$ port -q contents szip
  /opt/local/include/ricehdf.h
  /opt/local/include/szip_adpt.h
  /opt/local/include/szlib.h
  /opt/local/lib/libsz.2.dylib
  /opt/local/lib/libsz.a
  /opt/local/lib/libsz.dylib
```

I ask, as configuring without trace mode, does result in opportunistic use of `szip`. And when that happens, it also appears to disable szip-based encoding:

```
Features:
[...]
I/O filters (external): deflate(zlib),szip(no encoder)
```

Versus configuring in trace mode, which prevents use of that port:

```
Features:
[...]
I/O filters (external): deflate(zlib),szip(encoder)
```

Reply:
> For `--with-szlib`, was the intent to specify an invalid path, to sidestep opportunistic use of port `szip`?

No.  This was done to substitute **[libaec](https://ports.macports.org/port/libaec/details/)** for **[szip](https://ports.macports.org/port/szip/details/)**, to avoid a restrictive license.  Libaec includes a built-in replacement szip library.  This was changed a few months ago in Macports in #18775, following [upstream instructions](https://portal.hdfgroup.org/display/HDF5/Szip+Compression+in+HDF+Products).

This part of the portfile looks correct to me.